PurchasesCoreSwift/Networking/Backend.swift

```diff
@@ -0,0 +1,614 @@
+        }
+    }
+
+    @objc(getIntroEligibilityForAppUserID:receiptData:productIdentifiers:completion:)
+    public func getIntroEligibility(appUserID: String,
+                                    receiptData: Data,
+                                    productIdentifiers: [String],
+                                    completion: @escaping IntroEligibilityResponseHandler) {
+        guard productIdentifiers.count > 0 else {
+            completion([:])
+            return
+        }
+
+        if receiptData.count == 0 {
+            if SystemInfo.isSandbox {
+                Logger.appleWarning(Strings.receipt.no_sandbox_receipt_intro_eligibility)
+            }
+
+            var eligibilities: [String: IntroEligibility] = [:]
+
+            for productID in productIdentifiers {
+                eligibilities[productID] = IntroEligibility(eligibilityStatus: .unknown)
+            }
+
+            completion(eligibilities)
+            return
+        }
+
+        let fetchToken = receiptData.base64EncodedString(options: .lineLength64Characters)
+        let escapedAppUserID = escapedAppUserID(appUserID: appUserID)
+        let path = "/subscribers/\(escapedAppUserID)/intro_eligibility"
+        let body: [String: Any] = ["product_identifiers": productIdentifiers,
+                                   "fetch_token": fetchToken]
+
+        httpClient.performPOSTRequest(serially: true,
+                                      path: path,
+                                      requestBody: body,
+                                      headers: authHeaders) { statusCode, theResponse, error in
+            var response = theResponse
+            if statusCode >= HTTPStatusCodes.redirect.rawValue || error != nil {
+                response = [:]
+            }
+
+            guard let response = response else {
+                // missing response, we don't know eligibility for any of the identifiers.
+                let unknownEligibilities = [IntroEligibility](repeating: IntroEligibility(eligibilityStatus: .unknown),
+                                                              count: productIdentifiers.count)
+                let productIdentifiersToEligibility = zip(productIdentifiers, unknownEligibilities)
+                completion(Dictionary(uniqueKeysWithValues: productIdentifiersToEligibility))
+                return
+            }
+
+            var eligibilities: [String: IntroEligibility] = [:]
+            for productID in productIdentifiers {
+                let status: IntroEligibilityStatus
+
+                if let e = response[productID] as? NSNumber {
+                    status = e.boolValue ? .eligible : .ineligible
+                } else {
+                    status = .unknown
+                }
+
+                eligibilities[productID] = IntroEligibility(eligibilityStatus: status)
+
+            }
+            completion(eligibilities)
+        }
+    }
+
+    private func handleLogin(response: [String: Any]?,
```

Comment:
since there are so many `private` methods, they could be extracted to a `private extension`. WDYT?

Reply:
Love it.